internal/querycoord/cluster.go

```diff
@@ -300,7 +300,12 @@ func (c *queryNodeCluster) WatchDmChannels(ctx context.Context, nodeID int64, in
 
 			old, ok := c.clusterMeta.getDmChannel(info.ChannelName)
 			if ok {
+				removeFromSlice(old.NodeIds, in.OfflineNodeID)
```

Comment:
dmChannelInfos should be upated within cluster meta after triggering load balance of node down type. it seems not reasonable that remove offline node ids from a temporarily variable here.

Reply:
update logic problem: this pr means to remove offline node in time, when append new online node to dmChannel's nodes list.